src/backend/joanie/core/utils/contract.py

```diff
@@ -162,3 +163,41 @@ def order_has_organization_owner(order) -> bool:
     return OrganizationAccess.objects.filter(
         organization=order.organization, role=enums.OWNER
     ).exists()
+
+
+def get_signature_references(organization_id: str):
+    """
+    Returns all the backend references for the signature where the organization's
+    signature still remains.
+    """
+    return (
+        Order.objects.filter(
+            state=enums.ORDER_STATE_VALIDATED,
+            contract__student_signed_on__isnull=False,
+            contract__organization_signed_on__isnull=True,
+            organization_id=organization_id,
+        )
+        .values_list("contract__signature_backend_reference", flat=True)
+        .order_by("contract__signature_backend_reference")
+        .distinct()
+        .iterator()
+    )
```

Comment:
IMO we should retrieve submitted contracts not only one already signed by the student.

```suggestion
    return (
        Contract.objects.filter(
            submitted_for_signature_on__isnull=False,
	    order__state=enums.ORDER_STATE_VALIDATED
            order__organization_id=organization_id,
        )
        .values_list("signature_backend_reference", flat=True)
        .order_by("signature_backend_reference")
        .distinct()
        .iterator()
    )
```

Reply:
I thought that with the endpoint `submit_for_signature`, it would trigger the creation of the contract from the validated order, because we return the `invitation_link` at this specific time. 

With that in mind, I thought the student would always sign document as soon as the contract is created. 
I will apply your recommendation.